### ISSUE TITLE
[Bug] Save/Load functions do not work as expected

### ISSUE THREAD

**Issue by yangysc** [DLR-RM/stable-baselines3#449]
### 🐛 Bug

My personal task was trained well, but when I save it and load and test it, the model seems to be untrained. So I tested the cartpole examples. 

Why the evaluation after training the model is not consistent with the training info `ep_len_mean` and `ep_rew_mean` of `rollout` ? And why the untrained model can give the reward around `mean_reward=457.70 +/- 126.9` ?

How should we evaluate the trained model? My example is exactly based on the offical [example](https://colab.research.google.com/github/Stable-Baselines-Team/rl-colab-notebooks/blob/sb3/saving_loading_dqn.ipynb#scrollTo=pUWGZp3i9wyf) about saving and loading.

Thanks in advance!

### To Reproduce

Steps to reproduce the behavior.

```python
from stable_baselines3 import PPO
from stable_baselines3.common.evaluation import evaluate_policy
# seed
seed = 0
np.random.seed(seed)
th.manual_seed(seed)

env = gym.make("CartPole-v1")
eval_env = gym.make("CartPole-v1")

env.seed(seed)
eval_env.seed(0)

model = PPO("MlpPolicy", "CartPole-v1", verbose=1)

mean_reward, std_reward = evaluate_policy(model, eval_env, n_eval_episodes=10, deterministic=True)
print(f"mean_reward={mean_reward:.2f} +/- {std_reward}")
#
model.learn(total_timesteps=10000)
#
#
# # Save the agent
# model.save("CartPole")
# del model  # delete trained model to demonstrate loading
# model = PPO("MlpPolicy", env, verbose=1)

# model = PPO.load("CartPole")
mean_reward, std_reward = evaluate_policy(model, eval_env, n_eval_episodes=10, deterministic=True)

print(f"mean_reward={mean_reward:.2f} +/- {std_reward}")
```

```bash
C:\ProgramData\Anaconda3\envs\ray\python.exe E:/sb3_recompute_adv/stable-baselines3/stable_baselines3/cartpole_result_no_recompute_adv.py
Using cpu device
Creating environment from the given name 'CartPole-v1'
Wrapping the env with a `Monitor` wrapper
Wrapping the env in a DummyVecEnv.
e:\sb3_recompute_adv\stable-baselines3\stable_baselines3\common\evaluation.py:65: UserWarning: Evaluation environment is not wrapped with a ``Monitor`` wrapper. This may result in reporting modified episode lengths and rewards, if other wrappers happen to modify these. Consider wrapping environment first with ``Monitor`` wrapper.
  warnings.warn(
mean_reward=457.70 +/- 126.9
---------------------------------
| rollout/           |          |
|    ep_len_mean     | 22.5     |
|    ep_rew_mean     | 22.5     |
| time/              |          |
|    fps             | 2546     |
|    iterations      | 1        |
|    time_elapsed    | 0        |
|    total_timesteps | 2048     |
---------------------------------
-----------------------------------------
| rollout/                |             |
|    ep_len_mean          | 27          |
|    ep_rew_mean          | 27          |
| time/                   |             |
|    fps                  | 1703        |
|    iterations           | 2           |
|    time_elapsed         | 2           |
|    total_timesteps      | 4096        |
| train/                  |             |
|    approx_kl            | 0.008848501 |
|    clip_fraction        | 0.0925      |
|    clip_range           | 0.2         |
|    entropy_loss         | -0.686      |
|    explained_variance   | -0.00361    |
|    learning_rate        | 0.0003      |
|    loss                 | 7.04        |
|    n_updates            | 10          |
|    policy_gradient_loss | -0.0156     |
|    value_loss           | 53.6        |
-----------------------------------------
-----------------------------------------
| rollout/                |             |
|    ep_len_mean          | 37.4        |
|    ep_rew_mean          | 37.4        |
| time/                   |             |
|    fps                  | 1534        |
|    iterations           | 3           |
|    time_elapsed         | 4           |
|    total_timesteps      | 6144        |
| train/                  |             |
|    approx_kl            | 0.011284381 |
|    clip_fraction        | 0.0836      |
|    clip_range           | 0.2         |
|    entropy_loss         | -0.666      |
|    explained_variance   | 0.0815      |
|    learning_rate        | 0.0003      |
|    loss                 | 13.4        |
|    n_updates            | 20          |
|    policy_gradient_loss | -0.021      |
|    value_loss           | 36.3        |
-----------------------------------------
-----------------------------------------
| rollout/                |             |
|    ep_len_mean          | 48.5        |
|    ep_rew_mean          | 48.5        |
| time/                   |             |
|    fps                  | 1454        |
|    iterations           | 4           |
|    time_elapsed         | 5           |
|    total_timesteps      | 8192        |
| train/                  |             |
|    approx_kl            | 0.010004273 |
|    clip_fraction        | 0.0905      |
|    clip_range           | 0.2         |
|    entropy_loss         | -0.63       |
|    explained_variance   | 0.217       |
|    learning_rate        | 0.0003      |
|    loss                 | 22.6        |
|    n_updates            | 30          |
|    policy_gradient_loss | -0.0175     |
|    value_loss           | 56.8        |
-----------------------------------------
-----------------------------------------
| rollout/                |             |
|    ep_len_mean          | 62.9        |
|    ep_rew_mean          | 62.9        |
| time/                   |             |
|    fps                  | 1388        |
|    iterations           | 5           |
|    time_elapsed         | 7           |
|    total_timesteps      | 10240       |
| train/                  |             |
|    approx_kl            | 0.006279095 |
|    clip_fraction        | 0.062       |
|    clip_range           | 0.2         |
|    entropy_loss         | -0.596      |
|    explained_variance   | 0.259       |
|    learning_rate        | 0.0003      |
|    loss                 | 21          |
|    n_updates            | 40          |
|    policy_gradient_loss | -0.0154     |
|    value_loss           | 67.9        |
-----------------------------------------
mean_reward=362.20 +/- 131.3878228756379

Process finished with exit code 0


```

### System Info

Describe the characteristic of your environment:
 * pip install from source
 * CPU
 * Python version: 3.8
 * PyTorch version: 1.8.1
 * Gym version: 0.18.0
 * Versions of any other relevant libraries

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo (**required**)
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/) (**required**)
- [X] I have provided a minimal working example to reproduce the bug (**required**)

**Comment by yangysc** [comment]
Ok, I made a mistake. The save/load is all right. Close this.
